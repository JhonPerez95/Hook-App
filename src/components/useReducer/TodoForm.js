import React from 'react';
import PropTypes from 'prop-types';

import useForm from '../../hook/useForm';

const TodoForm = ({ addTodo }) => {
  const [formState, handleInputChange, resetInput] = useForm({
    description: '',
  });

  const handleSunmit = (e) => {
    e.preventDefault();

    if (description.trim().length <= 1) {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      description: description,
      done: false,
    };

    addTodo(newTodo);
    resetInput();
  };

  const { description } = formState;
  return (
    <>
      <h3>Add ToDo:</h3>
      <hr />
      <form onSubmit={handleSunmit}>
        <input
          type="text"
          className="form-control"
          name="description"
          placeholder="Comprar ..."
          autoComplete="off"
          value={description}
          onChange={handleInputChange}
        />
        <button
          className="btn btn-outline-primary mt-2 btn-block"
          type="submit"
        >
          Save ToDo
        </button>
      </form>
    </>
  );
};

TodoForm.propTypes = {
  addTodo: PropTypes.func.isRequired,
};

export default TodoForm;
