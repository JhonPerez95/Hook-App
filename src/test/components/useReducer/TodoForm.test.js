import React from 'react';
import { demoTodos } from './demoTodos';
import { shallow } from 'enzyme';
import TodoForm from '../../../components/useReducer/TodoForm';

describe('Pruebas al component <TodoForm/>', () => {
  const addTodo = jest.fn();
  const wrapper = shallow(<TodoForm addTodo={addTodo} />);

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('No debe llamar addToDo', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });
    expect(addTodo).toHaveBeenCalledTimes(0);
  });

  test('Debe llamar la funcion  addTodo con un argumento', () => {
    const value = 'Learn Git';
    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description',
      },
    });

    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });
    expect(addTodo).toHaveBeenCalled();
    expect(addTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      description: value,
      done: false,
    });

    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
