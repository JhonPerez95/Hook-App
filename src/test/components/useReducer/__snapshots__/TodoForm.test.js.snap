// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al component <TodoForm/> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Add ToDo:
  </h3>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Comprar ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-2 btn-block"
      type="submit"
    >
      Save ToDo
    </button>
  </form>
</Fragment>
`;
